backend/apps/kloudust/lib/cmd/scripts/powerOpVM.sh

```diff
@@ -3,17 +3,50 @@
 # Params
 # {1} - Domain / VM name
 # {2} - The operation to perform
+# {3} - Optional wait time before checking state (in seconds)
 
-NAME="{1}"
+VM_NAME="{1}"
 POWER_OP="{2}"
+WAIT_TIME="${3:-10}"
 
 function exitFailed() {
     echo Failed
     exit 1
 }
 
-printf "Power operating $NAME to $POWER_OP\n"
-if ! virsh $POWER_OP $NAME; then exitFailed; fi
+function backgroundCheckAndStartIfNeeded() {
+    local vm="$1"
+    local wait="$2"
 
-printf "\n\nPower operation $POWER_OP successful on $NAME\n"
-exit 0
+    (
+        sleep "$wait"
+        STATE=$(virsh domstate "$vm" 2>/dev/null)
+        if [[ "$STATE" == "shut off" ]]; then
+            echo "$vm is still shut off after $wait seconds. Starting it manually..."
+            if ! virsh start "$vm"; then
+                echo "Failed to start $vm after reboot"
```

Comment:
Did you test this? if you did, then did you realize this "echo" won't be captured as script has already ended for KD? Then what's the use of this?